esutil/bulk_indexer.go

```diff
@@ -425,73 +379,74 @@ func (w *worker) run() {
 		if w.bi.config.DebugLogger != nil {
 			w.bi.config.DebugLogger.Printf("[worker-%03d] Started\n", w.id)
 		}
-		defer w.bi.wg.Done()
+		defer func() {
+			w.ticker.Stop()
+			_ = w.flush(ctx)
```

Comment:
What's the reasoning for no propagating the error from flush ?

Reply:
I simply kept the logic as is, which is calling `bi.config.OnError()` if `flush()` returns an error. There is only one case where the flush error return is needed (see L407).
Indeed we could return `bool` instead of `error`. Though `error` might be useful in the future 🤷🏻‍♂️ .

Let me know what you prefer.

Reply:
Let's make that a boolean for the sake of consistency, we can always add back the error in the future if we need it.